crypto/asn1/asn1_gen.c

```diff
@@ -698,8 +698,13 @@ static ASN1_TYPE *asn1_str2type(const char *str, int format, int utype)
             atmp->value.asn1_string->data = rdata;
             atmp->value.asn1_string->length = rdlen;
             atmp->value.asn1_string->type = utype;
-        } else if (format == ASN1_GEN_FORMAT_ASCII)
-            ASN1_STRING_set(atmp->value.asn1_string, str, -1);
+        }
+        else if (format == ASN1_GEN_FORMAT_ASCII) {
+            ASN1_STRING_set(atmp->value.asn1_string, str, -1) {
```

Comment:
Should there be an `if` around the call to `ASN1_STRING_set()`? What am I missing?

Reply:
Yes, of course, this is my stupid mistake. Thank you.